test/streaming-client.spec.ts

```diff
@@ -0,0 +1,176 @@
+        });
+
+        it('adds event listener to httpResponse', () => {
+            // Setup
+            const res = sut.start();
+            let statusCode = null;
+            sut.on('httpResponse', code => {
+                statusCode = code;
+            });
+
+            // Act
+            mockClient.emit('httpResponse', {statusCode: 401});
+
+            // Assert
+            expect(statusCode).toBe(401);
+        });
+
+        it('adds event listener to connectFailed', () => {
+            // Setup
+            const res = sut.start();
+            let connectionError = null;
+            let expectedError = new Error("fake error")
+            sut.on('connectFailed', error => {
+                connectionError = error;
+            });
+            
+            // Act
+            mockClient.emit('connectFailed', expectedError);
+
+            // Assert
+            expect(connectionError).toBe(expectedError);
+        });
+
+        it('adds event listener to connection close', () => {
+            // Setup
+            const res = sut.start();
+            let closeCode = null;
+            let closeReason = null;
+            const expectedCloseCode = 1000;
+            const expectedCloseReason = "NormalClosure";
+            let mockConnection = new WebSocketConnection();
+            sut.on('close', (code, reason) => {
+                closeCode = code;
+                closeReason = reason;
+            });
+
+            // Act
+            mockClient.emit('connect', mockConnection);
+            mockConnection.emit('close', expectedCloseCode, expectedCloseReason);
+
+            // Assert
+            expect(closeCode).toBe(expectedCloseCode);
+            expect(closeReason).toBe(expectedCloseReason);
+        });
+
+        it('adds event listener to connection error', () => {
+            // Setup
+            const res = sut.start();
+            let connectionError = null;
+            const expectedError = new Error("fake connection error");
+            let mockConnection = new WebSocketConnection();
+            sut.on('error', error => {
+                connectionError = error;
+            });
+
+            // Act
+            mockClient.emit('connect', mockConnection);
+            mockConnection.emit('error', expectedError);
+
+            // Assert
+            expect(connectionError).toBe(expectedError);
+        });
+
+        it('emits connected event on connected message from server', () => {
+            // Setup
+            const res = sut.start();
+            let jobId = null;
+            const expectedJobId = "1";
+            let mockConnection = new WebSocketConnection();
+            sut.on('connect', response => {
+                jobId = response.id;
+            });
+
+            // Act
+            mockClient.emit('connect', mockConnection);
+            mockConnection.emit('message', 
+                {
+                    type: 'utf8', 
```

Comment:
Depending on your answer to the utf8 question above we might need another test for the non utf8 condition

Reply:
I don't think we need to test it as I don't think we should do anything in the case of non utf8 data

Reply:
We can test for the "do nothing" behavior.